stl/inc/generator

```diff
@@ -0,0 +1,592 @@
+// generator standard header
+
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#pragma once
+#ifndef _GENERATOR_
+#define _GENERATOR_
+#include <yvals_core.h>
+#if _STL_COMPILER_PREPROCESSOR
+
+#if !_HAS_CXX23
+_EMIT_STL_WARNING(STL4038, "The contents of <generator> are available only with C++23 or later.");
+#else // ^^^ !_HAS_CXX23 / _HAS_CXX23 vvv
+
+#include <coroutine>
+#include <exception>
+#include <xmemory>
+#ifdef __cpp_lib_byte
+#include <xpolymorphic_allocator.h>
+#endif // defined(__cpp_lib_byte)
+
+#pragma pack(push, _CRT_PACKING)
+#pragma warning(push, _STL_WARNING_LEVEL)
+#pragma warning(disable : _STL_DISABLED_WARNINGS)
+_STL_DISABLE_CLANG_WARNINGS
+#pragma push_macro("new")
+#undef new
+
+// TRANSITION, non-_Ugly attribute tokens
+#pragma push_macro("empty_bases")
+#undef empty_bases
+
+_STD_BEGIN
+
+struct alignas(__STDCPP_DEFAULT_NEW_ALIGNMENT__) _Aligned_block {
+    unsigned char _Pad[__STDCPP_DEFAULT_NEW_ALIGNMENT__];
+};
+
+template <class _Alloc>
+concept _Has_real_pointers = same_as<_Alloc, void> || is_pointer_v<typename allocator_traits<_Alloc>::pointer>;

```

Comment:
It it better to name it as `_Allocator_uses_raw_pointers`?

Reply:
The `same_as<_Alloc, void> ||` means this isn't really reusable, so I'm not too concerned about making the name discoverable. If anything, the `ispointer_v` term should be `_Allocator_uses_raw_pointers`, and this should be stuffed into a `generator`-specific detail namespace (`_Gen`?) along with some other entities that are not intended to be global but are shared by multiple parts of the `generator` impl.

Reply:
I wouldn't mind a detail namespace, but `_Gen` would be very shadowy.

Reply:
I chose `_Gen_detail`. There's a full description of the change in the OP.